data-prepper-plugins/mutate-string-processors/src/test/java/com/amazon/dataprepper/plugins/processor/mutatestring/SplitStringProcessorTests.java

```diff
@@ -58,11 +60,43 @@ void testSingleSplitProcessor(String message, List<String> splitMessage) {
         assertThat(splitRecords.get(0).getData().get("message", Object.class), equalTo(splitMessage));
     }
 
-    private SplitStringProcessorConfig.Entry createEntry(final String source, final String delimiter) {
-        return new SplitStringProcessorConfig.Entry(source, delimiter);
+    @Test
+    public void testBothDefinedThrowsError() {
+        when(config.getIterativeConfig()).thenReturn(Collections.singletonList(createEntry("message", ",", "a")));
+        when(config.getEntries()).thenReturn(Collections.singletonList(createEntry("message", ",", "a")));
+
+        assertThrows(IllegalArgumentException.class, () -> createObjectUnderTest());
+    }
+
+    @Test
+    public void testNeitherDefinedThrowsError() {
+        when(config.getIterativeConfig()).thenReturn(Collections.singletonList(createEntry("message", null, null)));
+        when(config.getEntries()).thenReturn(Collections.singletonList(createEntry("message", null, null)));
+
+        assertThrows(IllegalArgumentException.class, () -> createObjectUnderTest());
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(SplitStringArgumentsProvider.class)
+    void testDelimiterSplitProcessor(final String message, final List<String> splitMessage) {
+
+        when(config.getIterativeConfig()).thenReturn(Collections.singletonList(createEntry("message", null, ",")));
```

Comment:
It might be good to test this on a Java regex character - say `?` or `$` - instead of on `,`.

I think you will have to change the existing test as well. Or use two different `ArgumentsProvider`. I'm good with either approach.